### PR TITLE
added tests for healthcheck

### LINKED ISSUE #3988: [mesheryctl] Increase code coverage in `mesheryctl` upto 80%
#### Current Behavior

The current code coverage is less than 30% check here https://app.codecov.io/gh/meshery/meshery

#### Expected Behavior

Increase this code coverage up to 80%, this time!

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Freedisch**
**Notes for Reviewers**

This PR fixes #3988

PS: more tests for others files are coming in different PR

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by afzal442 — COMMENTED** [review]
thanks @Freedisch for your contribution. did you test it locally? there you got go-lint check failed, plz fix it `unusedresult: result of fmt.Errorf call not used (govet)`

**Comment by vishalvivekm** [comment]
@Freedisch 
Let's discuss this on Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by Freedisch** [comment]
@afzal442 , It should be alright now, there are some non-tested functions left I'll open another PR for them.

**Comment by Freedisch** [comment]
@Revolyssup

**Review by afzal442 — COMMENTED** [review]
Hi @Freedisch! I just reviewed it, nicely done but I came across a few of the func is not been covered. like https://github.com/meshery/meshery/blob/0cd2df56d398ecf68e252ae837c473affce8b179/mesheryctl/pkg/utils/healthcheck.go#L245-L333 wherever applicable. 🔋 

Edit: some non-tested functions?

File: mesheryctl/pkg/utils/healthcheck_test.go [hunk 1]
```diff
@@ -95,3 +97,88 @@ func TestCheckKubectlVersion(t *testing.T) {
...
+		{
+			minimum: [3]int{1, 12, 0},
+			current: [3]int{1, 12, 0},
+			expected: true,
+		},
+		{
+			minimum: [3]int{1, 11, 0},
+			current: [3]int{2, 13, 2},
+			expected: true,
+		},
```
Comment (hexxdump) [inline mesheryctl/pkg/utils/healthcheck_test.go:148]: can we have one negative test case here as well?
i.e. current < minimum version?

**Comment by leecalcote** [comment]
@afzal442 thanks for this review! 👏

File: mesheryctl/pkg/utils/healthcheck_test.go (hunk 1 above)
Comment (Freedisch) [reply mesheryctl/pkg/utils/healthcheck_test.go:148]: sure, I just added it

**Comment by Freedisch** [comment]
@afzal442 The remaining functions are using `c *meshkitkube.Client`  as arguments which makes them hardly testable right now

**Review by leecalcote — APPROVED** [review]
Let's give this a shot.

**Pull request merged**